### PR TITLE
Draggable simulator

### PR DESCRIPTION
**Pull request by ericnewcomer**

### CODE REVIEW

File: src/component/Flow.test.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ import { ConnectionEvent } from '../store';
 import { getFlowComponents, getGhostNode } from '../store/helpers';
 import { composeComponentTestUtils, composeDuxState, getSpecWrapper, setMock } from '../testUtils';
 import { merge, set, setTrue, dump } from '../utils';
+import toJson from 'enzyme-to-json';
```
Comment (ycleptkellan) [inline src/component/Flow.test.ts]: looks like we can lose this guy

File: src/component/Flow.test.ts [hunk 2]
```diff
@@ -209,14 +210,11 @@ describe(Flow.name, () => {
...
-                contactFields: [],
-                groups: [],
-                nodes: {}
-            });
-
+            // if we let jest string this one for us it fails
+            // presumably it's hitting an OOM due to circular references
+            // and reporting it as a RangeError
+            // https://github.com/nodejs/node-v0.x-archive/issues/14170
+            expect(JSON.stringify(wrapper.find('Simulator').props(), null, 1)).toMatchSnapshot();
```
Comment (ycleptkellan) [inline src/component/Flow.test.ts]: 😫

File: src/services/Plumber.ts [hunk 3]
```diff
@@ -107,23 +107,34 @@ export default class Plumber {
 
     public draggable(
         uuid: string,
-        start: Function,
-        drag: Function,
-        stop: Function,
-        beforeDrag: Function
+        start: Function = null,
```
Comment (ycleptkellan) [inline src/services/Plumber.ts]: just a thought for future development from a jsPlumb n00b: we might consider passing this method an options object so it's clearer on call which event handler a passed function is intended for.

Comment (ycleptkellan) [reply src/services/Plumber.ts]: of course, typing `plumberDraggable` would also help for those of us using vscode

File: src/component/Flow.test.ts (hunk 1 above)
Comment (ericnewcomer) [reply src/component/Flow.test.ts]: oops, right.

File: src/component/Flow.test.ts (hunk 2 above)
Comment (ericnewcomer) [reply src/component/Flow.test.ts]: 🤢 ... but oh well

File: src/services/Plumber.ts (hunk 3 above)
Comment (ericnewcomer) [reply src/services/Plumber.ts]: Feel like the plumber instance could alternatively just be context-ized

Comment (ycleptkellan) [reply src/services/Plumber.ts]: Ya perhaps that's a longer-term goal, migrating our services to the new context API?

Comment (ericnewcomer) [reply src/services/Plumber.ts]: Ya, perhaps :)

**Comment by ericnewcomer** [comment]
Double approval!! Thanks @ycleptkellan!

**Comment by ycleptkellan** [comment]
Rough internet sitch at the moment 😅

**Comment by ericnewcomer** [comment]
Don't be embarrassed by your approval exuberance. Your secret is safe with me.

**Pull request merged**